### PR TITLE
Design/tracing

### PR DESCRIPTION
**Pull request by chrusty**
First write up on Tracing.

### CODE REVIEW

File: design/platform/tracing.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+Tracing
+=======
+
+The time has come for distributed tracing. Fortunately we don't need to reinvent the wheel in 2020 because there are some great tools and useful standards at our disposal.
+
+
+Header injection
+----------------
+
+In the Micro platform a sensible idea would be to leave the tracing decision up to the client. In this way we can ensure that by default nothing is traced unless a decision had been specifically made. The Micro API would be a good starting point for tracing decisions. The client would be responsible to establishing the root span with appropriate metadata (eg originating API call or action), and all child spans would be aggregated beneath that (as the context is passed down).
```
Comment (domwong) [inline design/platform/tracing.md:10]: Makes sense. Really, the behaviour that I want to support most importantly is "I've noticed that API calls are failing consistently, I'm going to try myself and want to trace that call I make". So as long as I can initiate a traced rpc call from either curl or micro cli then that's the most important thing.

**Comment by domwong** [comment]
Sounds reasonable. Worth chatting to @arbarlow if he has any strong feelings on using Tempo vs Jaeger

**Comment by arbarlow** [comment]
Good doc! I think anything that supports opentracing and soon opentelemetry is our bet for sure.

Open telemetry seems like the direction to go since it's CNCF and such, so worth a spike.

I'd like to give Tempo a try, since it's only internal, it'd be nice to see if it's any good.

I think for end users, we might end up either wrapping Tempo or writing something a little custom

**Comment by arbarlow** [comment]
I've run Jaeger quite a few times, it's a pain to keep going with elasticsearch or cassandra

**Pull request merged**